src/plugins/summaryWidget/src/ConditionManager.js

```diff
@@ -0,0 +1,372 @@
+     * @param {Object} context A reference to a scope to use as the context for
+     *                         context for the callback function
+     */
+    ConditionManager.prototype.on = function (event, callback, context) {
+        if (this.supportedCallbacks.includes(event)) {
+            this.eventEmitter.on(event, callback, context || this);
+        }
+    };
+
+    /**
+     * Given a set of rules, execute the conditions associated with each rule
+     * and return the id of the last rule whose conditions evaluate to true
+     * @param {string[]} ruleOrder An array of rule IDs indicating what order They
+     *                             should be evaluated in
+     * @param {Object} rules An object mapping rule IDs to rule configurations
+     * @return {string} The ID of the rule to display on the widget
+     */
+    ConditionManager.prototype.executeRules = function (ruleOrder, rules) {
+        var self = this,
+            activeId = ruleOrder[0],
+            rule,
+            conditions;
+
+        ruleOrder.forEach(function (ruleId) {
+            rule = rules[ruleId];
+            conditions = rule.getProperty('trigger') === 'js' ?
+                rule.getProperty('jsCondition') : rule.getProperty('conditions');
+            if (self.evaluator.execute(conditions, rule.getProperty('trigger'))) {
+                activeId = ruleId;
+            }
+        });
+
+        return activeId;
+    };
+
+    /**
+     * Adds a field to the list of all available metadata fields in the widget
+     * @param {Object} metadatum An object representing a set of telemetry metadata
+     */
+    ConditionManager.prototype.addGlobalMetadata = function (metadatum) {
+        this.telemetryMetadataById.any[metadatum.key] = metadatum;
+        this.telemetryMetadataById.all[metadatum.key] = metadatum;
+    };
+
+    /**
+     * Adds a field to the list of properties for globally available metadata
+     * @param {string} key The key for the property this type applies to
+     * @param {string} type The type that should be associated with this property
+     */
+    ConditionManager.prototype.addGlobalPropertyType = function (key, type) {
+        this.telemetryTypesById.any[key] = type;
+        this.telemetryTypesById.all[key] = type;
+    };
+
+    /**
+     * Given a telemetry-producing domain object, associate each of it's telemetry
+     * fields with a type, parsing from historical data.
+     * @param {Object} object a domain object that can produce telemetry
+     * @return {Promise} A promise that resolves when a telemetry request
+     *                   has completed and types have been parsed
+     */
+    ConditionManager.prototype.parsePropertyTypes = function (object) {
+        var telemetryAPI = this.openmct.telemetry,
+            key,
+            type,
+            self = this;
+
+        self.telemetryTypesById[object.identifier.key] = {};
+        return telemetryAPI.request(object, {}).then(function (telemetry) {
```

Comment:
I could not get this to work, maybe leave this change for v1.1?

Reply:
They're optional parameters, not all backends will support them (SWG does not, from memory). The request parameters should at least be set though, so that backends that do support them can minimize the amount of data sent back.

Reply:
This should be documented.  I am pulling out a set of changes from the plot that adds this behavior to all example providers.